### PR TITLE
Delete removed entity from chado_bio_data_x

### PR DESCRIPTION
**Pull request by almasaeed2010**
Issue #652 

## Type(s) of Change(s)

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] API-specific change (fix or addition to an API function)
- [ ] Updates documentation (inline or markdown files)

## Description

As detailed in issue #652, deleting an entity using the GUI (or the `entity_delete` function), does not delete the corresponding record in the `chado_bio_data_x` table. This PR uses the `hook_entity_delete` hook to act on the deletion and remove the record from the other  table as well.

## Testing?

1. Before pulling this branch, create an organism entity and find the new record in `chado_bio_data_x`, where `x` is the `bundle_id` for `organism`
1. Delete the newly created organism using the GUI and you'll notice that the record still exists in the `chado_bio_data_x` table but does not exist in `tripal_entity`.
1. Now pull this branch and repeat the steps.
1. The record should no longer exist in the bundle table.

## Additional Notes (if any):

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: tripal_chado/includes/tripal_chado.entity.inc [hunk 1]
```diff
@@ -142,9 +142,13 @@ function tripal_chado_entity_update($entity, $type) {
...
 function tripal_chado_entity_delete($entity, $type) {
-  if ($type == 'TripalEntity') {
-
+  if ($type !== 'TripalEntity') {
+    return;
   }
+
+  // Delete the removed entity from the corresponding chado_bio_data_x table
+  $bundle = $entity->bundle;
+  db_delete("chado_{$bundle}")->condition('entity_id', $entity->id)->execute();
```
Comment (almasaeed2010) [inline tripal_chado/includes/tripal_chado.entity.inc:151]: Do we need a check here to verify the table exists? Or can we safely assume that it will exist?

**Comment by chunhuaicheng** [comment]
I tested the code and it did delete the entry from 'chado_bio_data_x'. However, I think it should delete the corresponding chado record as well. Here is our use case:

Our curator uses the web interface to delete unwanted pubs that were imported by the Tripal pub importer. After she deletes the pubs, they still show up in the Tripal Pub Search (/find/publications) results, even after applying you fix.

**Comment by almasaeed2010** [comment]
Hello @chunhuaicheng, 
Thanks for testing!

I think this is something the PMC (@spficklin, @laceysanderson, @bradfordcondon) should decide on. I do agree though that we need a method to delete from chado but I am not sure if there are cases where we don't want that behavior when deleting entities.

Thanks!

**Comment by bradfordcondon** [comment]
Deleting a tripal entity, intentionally, does not delete the chado record (thats my recollection of this discussion i've had with stephen).

@spficklin @laceysanderson I think this is not always the desired behavior, for example in the case @chunhuaicheng describes.  I would go so far as to say that deleting the chado record should be the **default behavior**.  That said, maybe we need to support both?  Deleting the record brings up a dialogue that lets you specify you want to **also delete the associated chado record** ?

We probably need a separate issue for this discussion: I'll create one.

**Review by laceysanderson — APPROVED** [review]
I confirmed that the bud described did exist and that this PR fixed it. Thanks @almasaeed2010!

 - [x] Checked that there was a record in chado_bio_data_1 for all my entities
```sql
tripaldb=# SELECT * FROM chado_bio_data_1;
 mapping_id | entity_id | record_id | nid 
------------+-----------+-----------+-----
          1 |         1 |         1 |    
          2 |         2 |         2 |    
          7 |         7 |       309 |    
(3 rows)
```

- [x] showed that after deleting one, I still have the `chado_bio_data_1` record but no record in `tripal_entity`

```sql
tripaldb=# SELECT * FROM chado_bio_data_1;
 mapping_id | entity_id | record_id | nid 
------------+-----------+-----------+-----
          1 |         1 |         1 |    
          2 |         2 |         2 |    
          7 |         7 |       309 |    
(3 rows)

tripaldb=# SELECT * FROM tripal_entity;
 id |     type     |   bundle   | term_id |      title      | uid | status |  created   |  changed   
----+--------------+------------+---------+-----------------+-----+--------+------------+------------
  1 | TripalEntity | bio_data_1 |       1 | Lens culinaris  |   1 |      1 | 1537117504 | 1537117517
  2 | TripalEntity | bio_data_1 |       1 | Lens ervoides   |   1 |      1 | 1537117530 | 1537117539
(2 rows)

tripaldb=# \q
```

- Checked out the branch

```bash
$ git checkout 652-tv3-entity_delete_fixBranch 652-tv3-entity_delete_fix set up to track remote branch 652-tv3-entity_delete_fix from origin.
Switched to a new branch '652-tv3-entity_delete_fix'

$ drush cc all
'all' cache was cleared in /var/www/dev/tripal#default                                         [success]
```

- [x] showed that deleting now, deletes both connections.

```sql
tripaldb=# SELECT * FROM tripal_entity;
 id |     type     |   bundle   | term_id |      title      | uid | status |  created   |  changed   
----+--------------+------------+---------+-----------------+-----+--------+------------+------------
  1 | TripalEntity | bio_data_1 |       1 | Lens culinaris  |   1 |      1 | 1537117504 | 1537117517
(1 row)

tripaldb=# SELECT * FROM chado_bio_data_1;
 mapping_id | entity_id | record_id | nid 
------------+-----------+-----------+-----
          1 |         1 |         1 |    
          7 |         7 |       309 |    
(2 rows)
```

**Comment by laceysanderson** [comment]
This PR is less than two lines and easily tested. As such, I've marked it needing **only 1 review**. Since I have provided that review, I'm merging this PR. Thank you @almasaeed2010!!!

**Pull request merged**